torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -790,12 +790,27 @@ void InitXlaModuleBindings(py::module m) {
       m, "XlaShardingSpec")
       .def(py::init([](at::Tensor tensor, const py::list& tile_assignment,
                        const py::list& group_assignment,
-                       const py::list& replication_groups, int sharding_type) {
+                       const py::list& replication_groups, int sharding_type,
+                       bool minibatch) {
+        xla::Shape tensor_shape =
+            CreateComputationShapeFromTensor(tensor, nullptr);
+        if (minibatch) {
+          int num_local_devices =
+              runtime::GetComputationClient()->GetLocalDevices().size();
+          int num_global_devices =
+              runtime::GetComputationClient()->GetAllDevices().size();
+          XLA_CHECK(tile_assignment.size() == num_global_devices)
```

Comment:
Naive question: how does this guarantee that only the batch dimension is sharded?

Reply:
In order to shard the input only along the batch axis. User would create a mesh with all the devices in the first dimension ( which will create a tile assignment like an array of all devices)